Tests/SwiftSyntaxTest/SymbolNameTests.swift

```diff
@@ -0,0 +1,358 @@
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import XCTest
+
+final class SymbolNameTests: XCTestCase {
+
+  // MARK: - Functions
+
+  func testFunctionNameWithoutParameters() {
+    assertSymbolName(ofFunction: "func name() {}", expecting: "name()")
+  }
+
+  func testFunctionNameWithSingleNamedParameter() {
+    assertSymbolName(ofFunction: "func name(one: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithSingleUnnamedParameter() {
+    assertSymbolName(ofFunction: "func name(_: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(one two: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithExplicitExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(one _: Int) {}", expecting: "name(one:)")
+  }
+
+  func testFunctionNameWithImplicitExternalParameterName() {
+    assertSymbolName(ofFunction: "func name(_ two: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithOnlyAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(_ _: Int) {}", expecting: "name(_:)")
+  }
+
+  func testFunctionNameWithMultipleNamedParameters() {
+    assertSymbolName(ofFunction: "func name(one two: Int, three: Int) {}", expecting: "name(one:three:)")
+  }
+
+  func testFunctionNameWithAllParametersHavingExternalNames() {
+    assertSymbolName(ofFunction: "func name(one two: Int, three four: Int) {}", expecting: "name(one:three:)")
+  }
+
+  func testFunctionNameWithMixedNamedAndAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(one two: Int, _ four: Int) {}", expecting: "name(one:_:)")
+  }
+
+  func testFunctionNameWithAllAnonymousParameters() {
+    assertSymbolName(ofFunction: "func name(_ two: Int, _ four: Int) {}", expecting: "name(_:_:)")
+  }
+
+  func testFunctionNameWithBackticks() {
+    assertSymbolName(ofFunction: "func `name`() {}", expecting: "`name`()")
+  }
+
+  func testFunctionNameWithParameterNameInBackticks() {
+    assertSymbolName(ofFunction: "func name(`one`: Int) {}", expecting: "name(`one`:)")
+  }
+
+  // FIXME: Does this make sense?
+  func testOperatorFunctionName() {
+    assertSymbolName(ofFunction: "func == (one: Int, two: Int)", expecting: "==(one:two:)")
+  }
+
+  private func assertSymbolName(
+    ofFunction function: DeclSyntax,
+    expecting expectedSymbolName: String,
+    file: StaticString = #filePath,
+    line: UInt = #line
+  ) {
+    guard let functionDecl = function.as(FunctionDeclSyntax.self) else {
+      XCTFail("Expected function declaration not found.", file: file, line: line)
+      return
+    }
+
+    XCTAssertEqual(functionDecl.symbolName, expectedSymbolName, file: file, line: line)
+  }
```

Comment:
I’d prefer to not interleave these `assert` functions with the test cases. I’d make them top-level functions at the start or the end of the file.

Reply:
Works for me, moved.